examples/translation/t5/evaluate_wmt.py

```diff
@@ -27,67 +25,104 @@ def generate_translations(lns, output_file_path, model_size, batch_size, device)
+            attention_mask = dct["attention_mask"].to(device)
 
-        translations = model.generate(input_ids=input_ids, attention_mask=attention_mask)
-        dec = [tokenizer.decode(g, skip_special_tokens=True, clean_up_tokenization_spaces=False) for g in translations]
+            translations = model.generate(
+                input_ids=input_ids, attention_mask=attention_mask
+            )
+            dec = [
+                tokenizer.decode(
+                    g, skip_special_tokens=True, clean_up_tokenization_spaces=False
+                )
+                for g in translations
+            ]
 
-        for hypothesis in dec:
-            output_file.write(hypothesis + "\n")
-            output_file.flush()
+            for hypothesis in dec:
+                output_file.write(hypothesis + "\n")
 
 
 def calculate_bleu_score(output_lns, refs_lns, score_path):
     bleu = corpus_bleu(output_lns, [refs_lns])
     result = "BLEU score: {}".format(bleu.score)
-    score_file = Path(score_path).open("w")
-    score_file.write(result)
+    with Path(score_path).open("w") as score_file:
+        score_file.write(result)
 
 
 def run_generate():
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "model_size",
+        "--model_size",
         type=str,
         help="T5 model size, either 't5-small', 't5-base', 't5-large', 't5-3b', 't5-11b'. Defaults to 't5-base'.",
         default="t5-base",
     )
     parser.add_argument(
-        "input_path", type=str, help="like wmt/newstest2013.en",
+        "input_path", type=str, help="like wmt/newstest2014.en",
     )
     parser.add_argument(
         "output_path", type=str, help="where to save translation",
     )
     parser.add_argument(
-        "reference_path", type=str, help="like wmt/newstest2013.de",
+        "reference_path", type=str, help="like wmt/newstest2014.de",
     )
     parser.add_argument(
         "score_path", type=str, help="where to save the bleu score",
     )
     parser.add_argument(
-        "--batch_size", type=int, default=16, required=False, help="batch size: how many to summarize at a time",
+        "--batch_size",
+        type=int,
+        default=16,
+        required=False,
+        help="batch size: how many to summarize at a time",
     )
     parser.add_argument(
-        "--no_cuda", default=False, type=bool, help="Whether to force the execution on CPU.",
+        "--no_cuda",
+        default=False,
+        type=bool,
+        help="Whether to force the execution on CPU.",
     )
 
     args = parser.parse_args()
-    args.device = torch.device("cuda" if torch.cuda.is_available() and not args.no_cuda else "cpu")
+    args.device = torch.device(
+        "cuda" if torch.cuda.is_available() and not args.no_cuda else "cpu"
+    )
 
     dash_pattern = (" ##AT##-##AT## ", "-")
 
-    input_lns = [x.strip().replace(dash_pattern[0], dash_pattern[1]) for x in open(args.input_path).readlines()]
```

Comment:
It's better to delete the three empty lists IMO, otherwise, we just propagate a silent error